data/src/main/java/com/linkedin/data/template/RecordTemplate.java

```diff
@@ -462,7 +463,7 @@ private Object obtainValueOrDefault(RecordDataSchema.Field field, GetMode mode)
    * @throws IllegalArgumentException if attempting to remove a mandatory field by setting it to null,
    *                                  see {@link SetMode#REMOVE_OPTIONAL_IF_NULL}.
    */
-  private boolean checkPutNullValue(RecordDataSchema.Field field, Object object, SetMode mode)
+  protected final boolean checkPutNullValue(RecordDataSchema.Field field, Object object, SetMode mode)
```

Comment:
Instead of changing this, providing another version of `putWrapped` and `putDirect` may be better?